zerver/lib/cache.py

```diff
@@ -534,3 +534,33 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         return wrapper
 
     return decorator
+
+def dict_to_items_tuple(user_function: Callable[..., Any]) -> Callable[..., Any]:
+    """Wrapper that converts any dict args to dict item tuples."""
+    def dict_to_tuple(k: Any) -> Any:
+        if isinstance(k, dict):
+            return tuple(k.items())
+        return k
```

Comment:
We should probably sort the keys to make this more stable

Reply:
Yep! Adding a sort.